lib/pry-doc/pry_ext/show_source_with_c_internals/code_fetcher.rb

```diff
@@ -0,0 +1,125 @@
+require 'fileutils'
+require_relative 'c_file'
+require_relative 'symbol_extractor'
+
+module Pry::CInternals
+  class CodeFetcher
+    include Pry::Helpers::Text
+
+    class << self
+      attr_accessor :ruby_source_folder
+    end
+
+    # normalized
+    def self.ruby_version() RUBY_VERSION.tr(".", "_") end
+
+    self.ruby_source_folder = File.join(File.expand_path("~/.pry.d"), "ruby-#{ruby_version}")
+
+    attr_reader :line_number_style
+    attr_reader :symbol_extractor
+
+    def initialize(line_number_style: nil)
+      @line_number_style = line_number_style
+      @symbol_extractor = SymbolExtractor.new(self.class.ruby_source_folder)
+    end
+
+    def fetch_all_definitions(symbol)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      "".tap do |result|
+        infos.count.times do |index|
+          result << fetch_first_definition(symbol, index).first << "\n"
+        end
+
+        return [result, full_path_for(infos.first.file)]
+      end
+    end
+
+    def fetch_first_definition(symbol, index=nil)
+      infos = self.class.symbol_map[symbol]
+      return unless infos
+
+      info = infos[index || 0]
+      code = symbol_extractor.extract(info)
+
+      "".tap do |result|
```

Comment:
Is there a necessity to use `.tap`? I don't see any benefit over using plain variables:

```rb
result = ''
result << "\n#{bold('From: ')}#{info.file} @ line #{info.line}:\n"
# ...
return [result, full_path_for(info.file)]
```

Besides, I find it unconventional to to use `.tap` and `return` from it (while being inside its context).

Reply:
@kyrylo yeah i was being weird, i'll get rid of it